### PR TITLE
runningFold method alongside tests

### PR DESCRIPTION
**Pull request by Wdestroier**
Hi,

I have added the runningFold method as requested in #151. Should the method stay in KtCollectionExtension?

The performed test is the same as the example present in Kotlin's documentation.

Reference: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/running-fold.html

### CODE REVIEW

**Comment by passsy** [comment]
[Kotlin impl](https://github.com/JetBrains/kotlin/blob/7060811c8b25678e7046f9fb569a8f0f1f782100/libraries/stdlib/common/src/generated/_Collections.kt#L2830-L2878)

File: lib/src/collection/kt_collection.dart [hunk 1]
```diff
@@ -56,6 +56,17 @@ extension KtCollectionExtensions<T> on KtCollection<T> {
     return sum as R;
   }
 
+  /// Returns a list containing the results of applying the given [operation]
+  /// function to each element in the original collection and the previous
+  /// accumulator value.
+  KtCollection<R> runningFold<R>(R initial, R Function(R, T) operation) {
```
Comment (passsy) [inline lib/src/collection/kt_collection.dart:62]: Change return type to `KtList`

Comment (passsy) [inline lib/src/collection/kt_collection.dart:62]: Create extension for `KtIterable`, not `KtCollection`

File: test/collection/collection_test.dart [hunk 2]
```diff
@@ -205,6 +205,18 @@ void testCollection(KtCollection<T> Function<T>() emptyCollection,
     });
   });
 
+  group("runningFold", () {
```
Comment (passsy) [inline test/collection/collection_test.dart:208]: Don't forget the `runningFoldIndexed` variant 😉

File: test/collection/collection_test.dart [hunk 3]
```diff
@@ -205,6 +205,18 @@ void testCollection(KtCollection<T> Function<T>() emptyCollection,
...
+    test("running folds a string list", () {
+      final strings = listOf<String>("a", "b", "c", "d");
```
Comment (passsy) [inline test/collection/collection_test.dart:210]: Add a test where the list is empty
